src/vm/amd64/pinvokestubs.S

```diff
@@ -130,24 +133,92 @@ NESTED_END VarargPInvokeGenILStub, _TEXT
 
 //
 // IN:
-// InlinedCallFrame (rcx) = pointer to the InlinedCallFrame data, including the GS cookie slot (GS cookie right 
+// InlinedCallFrame (rdi) = pointer to the InlinedCallFrame data, including the GS cookie slot (GS cookie right 
 //                          before actual InlinedCallFrame data)
 //
 //
-LEAF_ENTRY JIT_PInvokeBegin, _TEXT
-        // Not yet supported
-        int 3
+NESTED_ENTRY JIT_PInvokeBegin, _TEXT, NoHandler
+
+        alloc_stack         MIN_SIZE
+        save_reg_postrsp    r14, (MIN_SIZE + 0x08)
```

Comment:
I am not sure I understand this - you are saving the r14 into the caller's frame. Does the JIT reserve a slot there?

Reply:
The JIT does not resolve a slot. I copied this code from another stub, as an attempt to save the callee saved register.
What would be the proper way of doing this?

Reply:
What is the other stub on Unix where we have this? Unlike Windows, Unix doesn't have arguments save area in the caller's frame. So you need to save it in the slot of the current function. It seems that `push_nonvol_reg r14` as the first instruction and `pop_nonvol_reg r14` before the return would work. And you would need to make the MIN_SIZE multiple of 16 to preserve stack alignment. Alternatively, you can do:
```asm
alloc_stack (MIN_SIZE+8) 
save_reg_postrsp    r14, MIN_SIZE
```
and update the epilog of the function accordingly and again, make the MIN_SIZE multiple of 16.

Reply:
I'm not sure why MIN_SIZE was declared as 0x28 in other locations. Can this be smaller? Can it just be 0x10 here since I only need to save one register?